src/Model/Table/IncidentsTable.php

```diff
@@ -438,6 +442,30 @@ protected function _getSimpleVersion($versionString, $versionLength)
     }
 
     /**
+     * Returns the version string stripped of
+     * 'deb', 'ubuntu' and other suffixes
+     *
+     * @param string $versionString phpMyAdmin version
+     *
+     * @return string stripped phpMyAdmin version
+     */
+    protected function _getStrippedPmaVersion($versionString)
+    {
+        $allowedRegexp = '/^(\d+)(\.\d+){0,3}(\-.*)?/';
+        $matches = array();
+
+        // Check if $versionString matches the regexp
+        // and store the matched strings
+        if (preg_match($allowedRegexp, $versionString, $matches)) {
+            return $matches[0];
+        }
+
+        // If $versionString does not match the regexp at all,
+        // leave it as it is
+        return $versionString;
```

Comment:
I'd prefer to have the sanitization code just once, used both in validation of saved incidents and migration.